src/main.cpp

```diff
@@ -11,19 +11,19 @@
 #include "render.hpp"
 
 // Function to save image data in ppm format
-template <int width, int height> void save_image(std::array<color, (width * height)> & fb_data) {
+void save_image(int width, int height, auto& fb_data) {
```

Comment:
Ah yes, I forgot `auto` can work in functions and not only lambdas since C++20 with concepts